runelite-client/src/main/java/net/runelite/client/plugins/timetracking/farming/PatchImplementation.java

```diff
@@ -2603,6 +2605,137 @@ PatchState forVarbitValue(int value)
+				{
+					// Compost bin[Take,Examine,Dump] 30502,30503,30504,30505,30506,30507,30508,30509,30510,30511,30512,30513,30514,30515,30516
+					return new PatchState(Produce.ULTRACOMPOST, CropState.HARVESTABLE, value - 176);
+				}
+				return null;
+			}
+		},
+	GIANT_COMPOST(Tab.SPECIAL, "Giant Compost Bin")
+		{
+			@Override
+			PatchState forVarbitValue(int value)
+			{
+				if (value == 0)
+				{
+					// Big compost bin[Examine] 33762
+					return new PatchState(Produce.GIANT_COMPOST_BIN, CropState.EMPTY, 0);
+				}
+				if (value >= 1 && value <= 15)
+				{
+					// Big compost bin[Examine,Dump] 33763..33777
+					return new PatchState(Produce.GIANT_COMPOST, CropState.FILLING, value - 1);
+				}
+				if (value >= 16 && value <= 30)
+				{
+					// Big compost bin[Take,Examine,Dump] 33795..33809
+					return new PatchState(Produce.GIANT_COMPOST, CropState.HARVESTABLE, value - 16);
+				}
+				if (value >= 33 && value <= 47)
+				{
+					// Big compost bin[Examine,Dump] 33825..33839
+					return new PatchState(Produce.GIANT_SUPERCOMPOST, CropState.FILLING, value - 33);
+				}
+				if (value >= 48 && value <= 62)
+				{
+					// Big compost bin[Take,Examine,Dump] 33857..33871
+					return new PatchState(Produce.GIANT_SUPERCOMPOST, CropState.HARVESTABLE, value - 48);
+				}
+				if (value >= 63 && value <= 77)
+				{
+					// Big compost bin[Examine,Dump] 33778..33792
+					return new PatchState(Produce.GIANT_COMPOST, CropState.FILLING, 15 + value - 63);
+				}
+				if (value >= 78 && value <= 92)
+				{
+					// Giant compost bin[Take,Examine,Dump] 33810..33824
+					return new PatchState(Produce.GIANT_COMPOST, CropState.HARVESTABLE, 15 + value - 78);
+				}
+				if (value == 93)
+				{
+					// Giant compost bin[Open,Examine,Dump] 33794
+					return new PatchState(Produce.GIANT_COMPOST, CropState.GROWING, Produce.GIANT_COMPOST.getStages() - 1);
+				}
+				if (value >= 97 && value <= 99)
+				{
+					// Giant compost bin[Open] 33855,33855
+					return new PatchState(Produce.GIANT_SUPERCOMPOST, CropState.GROWING, value - 97); // Once closed, starts rotting (super compost)
+				}
+				if (value >= 100 && value <= 114)
+				{
+					// Giant compost bin[Take,Examine,Dump] 33872..33886
+					return new PatchState(Produce.GIANT_SUPERCOMPOST, CropState.HARVESTABLE, 15 + value - 100);
+				}
+				if (value >= 127 && value <= 128)
+				{
+					// Giant compost bin[Open,Examine,Dump] 33793,33793
+					return new PatchState(Produce.GIANT_COMPOST, CropState.GROWING, value - 127);
+				}
+				if (value >= 161 && value <= 175)
+				{
+					// Giant compost bin[Examine,Dump] 33840..33854
+					return new PatchState(Produce.GIANT_SUPERCOMPOST, CropState.FILLING, 15 + value - 161); // 161 means there are 16 items
+				}
+				if (value >= 176 && value <= 205)
+				{
+					// Giant compost bin[Take,Examine,Dump] 33957..33986
+					return new PatchState(Produce.GIANT_ULTRACOMPOST, CropState.HARVESTABLE, value - 176);
+				}
+				return null;
```

Comment:
what about 207-238?

Reply:
I didn't observe those enum values. Is there something indicating those varbits should be mapped? I was playing around with the setvarb command earlier and there were other varbits that seemed to map to similar states but again, they were never observed in my testing over a week of making compost.

Reply:
They have what looks like valid states in the object config so they should be mapped. Just take a guess, if its actually used and wrong we can fix it later.

Reply:
Yes you were right they correspond to rotten tomatoes.